### PR TITLE
Add 0.21 release notes

### PR DESCRIPTION
**Pull request by jarrodmillman**
Waiting on #6764.

### CODE REVIEW

**Review by stefanv — APPROVED** [review]

**Review by stefanv — APPROVED** [review]

File: doc/source/release_notes/release_0.21.rst
Comment (lagru) [inline doc/source/release_notes/release_0.21.rst:22]: I'll preface this with the statement that I won't block this. But I don't agree with this particular way of doing the release notes and without time-pressure or the feeling that I may be missing something I would veto this. I'm sorry for not speaking up earlier on this, but from previous discussions I didn't think that the goal was to be this terse in the notes. I get the argument that creating elaborate release notes is work and shouldn't be taken to an extreme. But I also worry that this may be going to much into the other extreme. 

I think release notes are documentation that is useful to us and to users who are curious about changes between releases and want to discover new features and be aware of deprecated ones. How else are they supposed to find out? And with regard to users, I really don't want to make them read PRs. Especially because those may contain anything from outdated descriptions to discussions that are to large to read. PRs also don't necessarily correspond to a good bullet point in the release notes. Changes belonging to the same effort may be split across PRs or PRs may update independent things that deserve independent entries in the notes. 

I think the line I'm commenting on here is a good example of my critique. What is a user supposed to learn from "Unify pseudo-random seeding interface"? There is no mention of refactoring to `rng` and the actual change is split among two PRs of which the latter isn't mentioned here (https://github.com/scikit-image/scikit-image/pull/6924). At this level, release notes cease being useful and are just a fancy reference list of development steps that may or may not have useful titles. 

I also want to point out that this is deviating from what SciPy, NumPy and other packages in the ecosystem do. Is our changes / maintainer resources ratio really that much higher than theirs to warrant this trade-off? 

I'm not sure how time-critical you think this is but I'm willing to go over these over the weekend and try to arrive at a compromise if you are willing to wait for that. And as always, I mean this is the most constructive way possible and are of course open to arguments. :)

Comment (stefanv) [reply doc/source/release_notes/release_0.21.rst:22]: My preference is to have a tool that can extract slightly longer descriptions from the PR itself. You can then add a code block to the PR description for any PR that warrants some explanation, and those can be automatically put first in the release notes.

Such a tool does not exist yet.

In the mean time, we have to ask ourselves: where do we want to spend our time? Editing release notes, or improving the software. And the answer to that question may differ for individuals on the core team.

We are very, very far behind on handling issues and incoming PRs, so I, personally, would prefer to get a grip on that. I am not opposed to others spending time on the release notes, but it is an opportunity cost.

That said, in this specific case it is true that there is an API change afoot that warrants communication to the users. A single paragraph added under that PR should suffice.

Comment (jarrodmillman) [reply doc/source/release_notes/release_0.21.rst:22]: @lagru If you have time to work on the release this weekend, it would be great if you could track down the error blocking the release before focusing on the release notes. I suspect it is related to our test suite and may not be a real issue, but I am not sure.

Here is the error:
```
FAILED io/tests/test_imageio.py::TestSave::test_bool_array_save - ValueError: Unexpected warning: unclosed file <_io.BufferedReader name='/home/runner/.cache/scikit-image/main/data/no_time_for_that_tiny.gif'>
```

The odd thing to me is that `io/tests/test_imageio.py` doesn't reference `data/no_time_for_that_tiny.gif`.

The release is 23 days overdue and this is the second release that has been delayed for several weeks. We should discuss ways to make getting releases out more easily because currently it is very time-consuming.

I am going to merge this PR and make 0.21.0rc1 release today to see if we get more bug reports. Since main is failing the CI, I doubt we will be able to get 0.21.0 out before next Tuesday. We can discuss how much time to spend on the release notes during the developer call on Tuesday or next week at the summit.

Comment (lagru) [reply doc/source/release_notes/release_0.21.rst:22]: > My preference is to have a tool that can extract slightly longer descriptions from the PR itself.

I agree. I'll create a topic in our forum to get this discussion going. I probably should have done so in the first place and probably could have used a milder tone in my initial comment. Thank you anyway for engaging. :)

> Such a tool does not exist yet.

Might be something to discuss or even work on during the summit?

> In the mean time, we have to ask ourselves: where do we want to spend our time? Editing release notes, or improving the software. And the answer to that question may differ for individuals on the core team.

I consider release notes as part of and good practice for software. You are right though, the priority will be different for people.

> if you could track down the error blocking the release before focusing on the release notes.

I'll have a look. I think I noticed that particular bug in #6764 as well, but rerunning the job seemed to fix it. So it's probably one of those bugs that breaks the CI only sometimes?

> The release is 23 days overdue and this is the second release that has been delayed for several weeks. We should discuss ways to make getting releases out more easily because currently it is very time-consuming.

Thank you for working on these and pushing these through! I'm very much on board with this goal. My personal impression was that it wasn't delayed due to release notes. They were a bigger factor during for v0.20, though considering the size of that release I find that somewhat appropriate.

Comment (lagru) [reply doc/source/release_notes/release_0.21.rst:22]: See https://discuss.scientific-python.org/t/738.

Comment (crisluengo) [reply doc/source/release_notes/release_0.21.rst:22]: Excuse me butting in here.

I think you need to require each PR to add a line to the release notes. I actually thought this was already a requirement, given you have a file for it: https://github.com/scikit-image/scikit-image/blob/main/doc/source/release_notes/release_dev.rst

If you don’t merge any PRs that don’t add a line to that file, it will be much less work finalizing good release notes.

An additional benefit is that it is the author of the PR writing the summary, this is likely the best suited person to explain concisely what changed.

I think that release notes are essential. Looking through the list of merged PRs is not a good way to learn what has changed.

Comment (stefanv) [reply doc/source/release_notes/release_0.21.rst:22]: We're trying to streamline our release process a bit. We agree that there should be release notes for certain PRs, and I think the preferred idea right now is to grab those directly from PRs, where they can be designated by a special header or code fence.

**Pull request merged**
